### PR TITLE
dockerTools: Add example of using NixOS' etc

### PR DESCRIPTION
**Pull request by roberth**
###### Description of changes

(cherry picked from commit 9b2af8673be82d48ce76c8c152de85ad921d26ba)

#148456 came up again; figured I'd try to make some progress again.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/docker/examples.nix [hunk 1]
```diff
@@ -582,6 +592,37 @@ rec {
...
+          # mode = "0755";
+        };
+      }));
+    in pkgs.dockerTools.streamLayeredImage {
+      name = "etc";
+      tag = "latest";
+      enableFakechroot = true;
+      fakeRootCommands = ''
+        mkdir -p /etc
+        ${nixosCore.config.system.build.etcActivationCommands}
```
Comment (roberth) [inline pkgs/build-support/docker/examples.nix:618]: We should have a function in `dockerTools` that just takes a module and runs the whole activation script, but I don't think the modules are in a state to support that yet. Until then, this is the best we can do.

File: nixos/tests/docker-tools.nix [hunk 2]
```diff
@@ -419,5 +419,10 @@ import ./make-test-python.nix ({ pkgs, ... }: {
             "docker rmi layered-image-with-path",
         )
 
+    with subtest("etc"):
+        docker.succeed("${examples.etc} | docker load")
```
Comment (06kellyjac) [inline nixos/tests/docker-tools.nix:423]: Does this not need to be

```suggestion
        docker.succeed("cat ${examples.etc} | docker load")
```

or

```suggestion
        docker.succeed("docker load -i ${examples.etc}")
```

?

**Review by 06kellyjac — COMMENTED** [review]
Nice

Just 1 question

File: nixos/tests/docker-tools.nix (hunk 2 above)
Comment (06kellyjac) [reply nixos/tests/docker-tools.nix:423]: ahh

**Pull request merged**